ctru-rs/src/applets/swkbd.rs

```diff
@@ -207,6 +213,20 @@ bitflags! {
     }
 }
 
+struct MessageCallbackData {
+    extra: *mut SwkbdExtra,
+    swkbd_shared_mem_ptr: *mut libc::c_void,
+}
```

Comment:
I only made this struct because `libctru` doesn't include a pointer to the shared memory block in `SwkbdExtra` like it ought to, and instead uses a `static` to pass the pointer around. I decided there's no reason we have to do the same in our implementation, I've never messed with any other APT message calls, so I'm not sure if this specific struct would be helpful outside of this specific context.